### PR TITLE
Parallelize tests

### PR DESCRIPTION
**Pull request by bkabrda**
### What does this PR do?

Turns on parallelization via forks when running tests. Works for recording/replaying as well.

### Review checklist
Please check relevant items below:
- [ ] This PR includes all [newly recorded cassettes](/DEVELOPMENT.md) for any modified tests.

- [ ] This PR does not rely on API client schema changes.
    - [ ] The CI should be fully passing.
- [ ] Or, this PR relies on API schema changes and this is a Draft PR that includes tests.
  - Note: CI shouldn't be run on this Draft PR, as its expected to fail without the corresponding schema changes.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -77,13 +77,18 @@
                 <version>2.19.1</version>
                 <configuration>
                     <runOrder>random</runOrder>
+                    <!-- We don't use threads, as that would complicate cassette recording, plus keystore used by tests isn't thread safe. -->
```
Comment (gzussa) [inline pom.xml:80]: 👍 Thanks for this comment!

File: src/test/java/com/datadog/api/v1/client/api/AwsIntegrationApiTest.java [hunk 2]
```diff
@@ -29,10 +27,21 @@
  * API tests for AwsIntegrationApi
  */
 public class AwsIntegrationApiTest extends V1ApiTest {
+    /*
```
Comment (gzussa) [inline src/test/java/com/datadog/api/v1/client/api/AwsIntegrationApiTest.java:30]: Great!

File: src/test/java/com/datadog/api/TestUtils.java [hunk 3]
```diff
@@ -136,8 +138,9 @@ public HttpURLConnection getConnection(URL url) throws IOException {
 
         protected static String TEST_API_KEY;
         protected static String TEST_APP_KEY;
+        protected static int WIREMOCK_PORT = 8080 + SUREFIRE_FORK - 1;
```
Comment (gzussa) [inline src/test/java/com/datadog/api/TestUtils.java:141]: why `-1`?

Comment (bkabrda) [reply src/test/java/com/datadog/api/TestUtils.java:141]: Heh, I think this is a remainder of some other attempts and it's not important. I'll delete the `- 1` to make it less confusing.

**Comment by bkabrda** [comment]
@jirikuncar yeah, I've been thinking that myself. I'll open another PR to do that.

**Pull request merged**